### DISCUSSION TITLE
Editing temporary list of items before POST to a REST data source

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#11979] [Help] [answered]
We have a REST API that supports CRUD for for accounts. Each account has a list of users with several attributes (like email, name, admin, etc). When creating a new account, how can we display an editable list of users in a form before posting? Is there any way to bind a repeater to a JavaScript array or a temporary table?

**Comment by ConorWebb96** [comment]
Hey @wmherold,

Yes this is something that can be done. I've broken down and example of mine below hopefully this will help you!

1. Heres my setup of my rest get request including the binding I'm using the fetch the specific record.
<img width="951" alt="Screenshot 2023-10-09 at 08 03 57" src="https://github.com/Budibase/budibase/assets/126772285/11d74425-c9e6-4209-9533-5b81a9da2cdb">
2. You can alter the id binding in the previous screen within the design area using the settings cog on the data provider.
<img width="1511" alt="Screenshot 2023-10-09 at 08 04 53" src="https://github.com/Budibase/budibase/assets/126772285/1440a265-b6c2-4686-82e9-05a5c24ab0bb">
3. When clicking the cog in the above screenshot a drawer should open that lets you configure the bindables values in your rest request. Heres how mine looks.
<img width="891" alt="Screenshot 2023-10-09 at 08 05 43" src="https://github.com/Budibase/budibase/assets/126772285/ae832ee1-9946-45a3-adeb-a49152f36a04">
4. Make sure your form component schema is set to custom.
<img width="100%" alt="Screenshot 2023-10-09 at 08 07 09" src="https://github.com/Budibase/budibase/assets/126772285/5c9e3fa4-48c0-42e0-b64e-d5200a64bcd1">
5. You will have to manually bind your field names to match the temp schema that your rest request generates. Below is an example of my id config.
<img width="100%" alt="Screenshot 2023-10-09 at 08 07 00" src="https://github.com/Budibase/budibase/assets/126772285/c941a2aa-ef09-4673-859e-e8fca1db6929">
6. If you are wanting to bind this up to a post request you should be able to use the the `**execute query**` action and bind up all the fields within your form using the bindings within the settings. (Similar approach to the data provider setup talked about earlier.)


Heres a demo of everything working. I've opted to set this up using url variables, this can also be done with side panel/nest forms inside the side panel.
![Screen Recording 2023-10-09 at 08 07 37](https://github.com/Budibase/budibase/assets/126772285/d4c73a48-52d6-47cc-9478-d6523b367b85)

I hope this helps.

**Comment by ghost** [comment]
Thank you for the detailed response, but I'm not sure I understand it. We want a form to create a new record ("account") that has child records ("admins"). Picture a form with some text fields and a table. The table has buttons to add & remove items. The form has a Save button that calls the the REST query to actually create the account along with the admins. 

The simplified json sent in the POST would look like this. 
```
{
    "accountName": "New Account",
    "admins": [
        {
            "id" : "123",
            "email" : "userA@foo.com"
        },
        {
            "id" : "456",
            "email" : "userB@foo.com"
        }
    ]
}
```

The question is how do we display the list of admins in the account form, and allow adding and removing admins without actually calling a REST endpoint each time? The account doesn't exist yet so we can't actually add & remove admins on the back end until it is created.

**Reply by ghost** [reply]
If this is not possible, we could split managing accounts and admins into two applications, or require an account is created before admins can be added & removed. But this would be an awkward workflow.

**Reply by ConorWebb96** [reply, marked as answer]
Oh, you wish to create a dynamic form. This currently isn't possible with Budibase heres the related [issue](https://github.com/Budibase/budibase/issues/5658).

You may be able to display the json information using something like the [super repeater plugin](https://github.com/poirazis/bb-component-SuperRepeater). 

I hope this helps.

**Reply by ghost** [reply]
That is helpful, thank you. I think we will plan on requiring a save first, but I will look into that plugin as a possible enhancement.
